src/ClassDiscovery.php

```diff
@@ -120,9 +121,9 @@ public static function setStrategies(array $strategies)
     /**
      * Append a strategy at the end of the strategy queue.
      *
-     * @param DiscoveryStrategy $strategy
+     * @param string $strategy Fully qualified class name to a DiscoveryStrategy
      */
-    public static function appendStrategy(DiscoveryStrategy $strategy)
+    public static function appendStrategy($strategy)
```

Comment:
and to test this, first setStrategies to empty or something that never works, then append the one that returns a mock client.